Roman

amymhaddad opened the pull request:
Please review this Roman Numeral solution. I need to get the letters to appear in reverse order, which I'm having trouble with. I've tried using sorted() and using a -1 index. 

For example, if given the number: 6, the roman numeral returns ['I','V'] instead of ['V','I'].

amymhaddad committed:
Adding roman numeral

amymhaddad committed:
Algorithm works for getting the value from the dictionary and reducing the given number by the amount found in the dictionary

amymhaddad committed:
Updated algorithm to check if number is a multiple of 5, and to reduce the number by 1 if not

amymhaddad committed:
Algorithm works on all integers to convert from int to roman numeral. Now need to get the letters to return in reverse order

amymhaddad committed:
Made multiples of five a function

amymhaddad committed:
Final changes for draft PR review

paulghaddad commented on roman_numeral/roman_numerals.py:
```diff
@@ -0,0 +1,40 @@
+        convert = []
+        reduced_num = number
+
+        if number in multiples_of_five():
```
Can't you just do: `number % 5 == 0`?

paulghaddad commented on roman_numeral/roman_numerals.py:
```diff
@@ -0,0 +1,40 @@
+        if number in multiples_of_five():
+            convert.append(roman_numeral_values[number])
+            reduced_num -= number
+        else:
```
So your question on the special case of `IV`...

so this is the third type of condition you could have. You are already dealing with the first two correctly. I'm going to leave it to you to figure out how to test for this condition, and then, how to deal with it. But start with the conditional test you'll need.

paulghaddad reviewed (commented):
Good work so far. Addressed a couple things but want you to think about how to deal with the IV condition.

amymhaddad committed:
Refactored finding the multiples of 5 to 1 line and removed an entire function

amymhaddad committed:
Updated roman so the roman numerals appear in the correct order

amymhaddad committed:
Updated first function to handle numbers that are multiples of 5, but larger than the dictionary multiples of 5

amymhaddad committed:
Updated roman to account for numbers that are not in dictionary and are larger than 5

amymhaddad committed:
Updates to draft PR

amymhaddad commented:
@paulghaddad I've updated this program. The order of the numbers is correct. I also updated the program to account for a range of numbers (ie, 6, 16, 15).

I added several try, excepts. Is this a good way of handling the different scenarios?

paulghaddad commented on roman_numeral/roman_numerals.py:
```diff
@@ -0,0 +1,52 @@
+
+        # First check: see if number is in dictionary. If so, return it, if not get the largest value from the dictionary
+        if number % 5 == 0:
+            try:
```
Look closely at lines 23-28 and see if there's duplication that you can eliminate...same with lines 34-41.

paulghaddad replied:
hint, you probably don't need to use the `try-except` here.

paulghaddad commented on roman_numeral/roman_numerals.py:
```diff
@@ -0,0 +1,52 @@
+                convert.append(roman_numeral_values[10])
+                reduced_num -= 10
+
+            return "".join(convert) + conversion(reduced_num)
```
Is it necessary to have `convert` as a list? It seems a string would be more than fine because you're ultimately converting it to a string on line 30 with `join` anyway.

paulghaddad reviewed (changes requested):
So I've made a first round of changes that will simplify your code and ultimately allow you to accomplish the base - 1 (ie IV) case. But I feel it's important to do this phase of refactoring first because it will make the subsequent refactoring easier. 

Nice work overall! This is not an easy one.
